src/components/add-transaction-dialog/add-transaction-dialog.js

```diff
@@ -1,28 +1,68 @@
 import template from "./add-transaction-dialog.html";
+
 import { MDCDialog } from "@material/dialog";
 import { MDCSelect } from "@material/select";
 import { MDCTextField } from "@material/textfield";
 import { MDCRadio } from "@material/radio";
 
+import { CustomSelectComponent } from "../custom-select/custom-select";
+
+import { AccountListService } from "../../services/account-service";
+import { TagListService } from "../../services/tag-service";
+
 export class AddTransactionComponent {
   constructor(mountPoint, props) {
     this.mountPoint = mountPoint;
     this.props = props;
   }
 
+  getStoredAccounts() {
+    AccountListService.get()
+      .then(accounts => this.showAccounts(accounts))
+      .catch(e => console.error(`get accounts: ${e.message}`));
+  }
+
+  showAccounts(accounts) {
+    this.accounts = accounts ? accounts : [];
+    this.accountSelect = new CustomSelectComponent(
+      this.accountSelectMountPoint,
+      {
+        type: "account",
+        items: this.accounts.map(item => item.name)
+      }
+    );
+    this.accountSelect.mount();
+  }
+
+  getStoredTags() {
+    TagListService.get()
+      .then(tags => this.showTags(tags))
+      .catch(e => console.error(`get tags: ${e.message}`));
+  }
+
+  showTags(tags) {
+    this.tagSelect = new CustomSelectComponent(this.tagSelectMountPoint, {
```

Comment:
you should mount component at dialog start in mountChildren -- yes there is no data at that time -- but it still should be mounted without data. And then when data arrives -- you should update this select